prototype2/common/Socket.cpp

```diff
@@ -17,6 +17,25 @@
 #define SEND_FLAGS 0
 #endif
 
+
+bool Socket::isValidIp(std::string ipAddress) {
+  struct sockaddr_in sa;
+  int result = inet_pton(AF_INET, ipAddress.c_str(), &(sa.sin_addr));
+  return result != 0;
+}
+
+std::string Socket::getHostByName(std::string &name) {
+  struct hostent * hp = gethostbyname(name.c_str());
+  if (hp == NULL) {
+    std::string ret;
+    ret = fmt::format("Unable to resolve hostname {}", name);
+    throw std::runtime_error(ret);
+  } else { // Just return the first entry
+    auto res = inet_ntoa(*(struct in_addr*)(hp->h_addr_list[0]));
```

Comment:
```suggestion
    auto res = inet_ntoa(*reinterpret_cast<in_addr*>(hp->h_addr_list[0]));
```
Use `reinterpret_cast`.